docu: add process description to readme (#5)
Miragon/hrworks-showcase: README.md

This project serves as a showcase for demonstrating the synchronization of times between Clockify and HRWorks using Camunda 7 and following the principles of the Hexagonal Architecture.

## The Process

<img alt="hrworks-bpmn-diagram" src="images/hrworks-timesync.png">

This BPMN diagram represents the **HRworks TimeSync Process** using the Camunda Modeler (exporter version 5.13.0).

The process involves synchronizing time-related data within the HRworks system. It consists of the following key steps:

1. **Start Event**: The process begins with a start event. This should be changed to a timer start event to trigger automatically at a specific time.

> 2. **Load Users Service Task**: This external service task named "Load Users" is executed asynchronously. It loads user data and related information. The loaded data includes users, employees, and workspace information.

Reviewer: Do we execute the Task async?

3. **Subprocess - Sync Time Entries for Users**:

>     - This subprocess is a multi-instance subprocess, executed asynchronously for each user.

Reviewer: ```suggestion
    - This subprocess is a multi-instance subprocess, executed asynchronously and in parallel for each user.
```

    - It contains several tasks:

>         - **Start Event**: This event initiates the subprocess for each user.

Reviewer: Start Event is not a task.

>         - **Load Time Entries Service Task**: This external service task loads time entry data for the user.

Reviewer: ```suggestion
        - **Load Time Entries Service Task**: This external service task loads time entry data for the user. When an error occurs, the boundary event is triggered.
```

        - **Sync Time Service Task**: Another external service task syncs the time entries for the user.

>         - **Notify User Task**: This task notifies the user about the process status. There is one error triggered by purpose in the code to demonstrate the notification user task.

Reviewer: ```suggestion
        - **Notify User Task**: This task notifies the user when a user's time could not be loaded successfully. There is one error triggered by purpose in the code to demonstrate the notification user task.
```

> 4. **End Events**:

Reviewer: Shouldn't the End Event of the subprocess also be listed under 3.?
